module/zfs/arc.c

```diff
@@ -7214,7 +7213,6 @@ arc_tempreserve_space(spa_t *spa, uint64_t reserve, uint64_t txg)
 	uint64_t spa_dirty_anon = spa_dirty_data(spa);
 	uint64_t rarc_c = arc_warm ? arc_c : arc_c_max;
 	if (total_dirty > rarc_c * zfs_arc_dirty_limit_percent / 100 &&
-	    anon_size > rarc_c * zfs_arc_anon_limit_percent / 100 &&
```

Comment:
It should be fine remove the check for anon_size, which is a part of total_dirty, right?